### PR TITLE
Add Project section to the schema 

### PR DESCRIPTION
**Pull request by ashes74**
For developers who want to highlight some of their projects:

Include highlights, technology used, teammate names. 

Example: 

"projects": [
       {
       "name": "Miss Direction",
       "summary": "A mapping engine that misguides you",
       "highlights": [ 
         "Won award at AIHacks 2016",
         "Built by all women team of newbie programmers"
       ],
       "tech": [
         "GoogleMaps", "Chrome Extension", "Javascript"
       ],
       "startDate": "2016-08-24",
       "endDate": "2016-08-24", 
       "website": "missdirection.io",
       "team": ["Gale Core", "Dani Yes"]

```
   }
```

  ],

Items left:
* [x] ```summary``` to ```description```
* [x] ```website``` to ```url```
* [x] add "format": "uri" to ```url```
* [x] example url as subdomain from example.com
* [x] add ```entity``` e.g. "greenpeace", "corporationXYZ"
* [x] add ```type``` e.g. "presentation", "conference"
* [x] add ```roles``` e.g. "speaker", "attendee"
* [x] remove html (optional)

### CODE REVIEW

**Comment by aloisdg** [comment]
Can you link relevant issues?

How `tech` is this relevant to non developers?

Thank you for your time though :)

**Comment by stp-ip** [comment]
Thanks for your time implementing this. For technology jobs that sounds reasonable, but we might have to change the projects implementation to conform more to the discussion in #201. 
Also team and especially tech are quite specific. 

What's the reasoning behind naming the team? It's your resume and not that of your teammates, so why push other names out there? Team information could easily live on the project website or within a longer description I assume, if really necessary.

If we decide to add something like tech, I would make the naming more generic. But couldn't that live within the highlights section: "Using modern technologies such as GoogleMaps, Chrome Extension and Javascript"
It seems easier to customize, less clutter from another set of keyword like things.

Another variant would be to use keywords as suggested within #201. So instead of tech the technology could be added as keywords, which would conform better to the other sections.

Thanks again and awesome that we are moving again. \o/

**Comment by ashes74** [comment]
Thanks for the feedback and links to existing issues, @stp-ip 

Good point on the team property. It would seem including personal role on the project would be a more valid property. 

"Keywords" can definitely replace "tech" to make projects more universal. 

I like the ideas you have here: https://github.com/jsonresume/resume-schema/issues/201#issuecomment-157872094

It captures all the things that inspired me to make this PR. What's needed to implement either of the numerous proposals for Projects?

**Comment by aloisdg** [comment]
> What's needed to implement 

AFAIK, feedback and a final decision.

**Comment by stp-ip** [comment]
I'm on the go the next week, but hope that I get some time afterwards to give tangible feedback.

**Comment by thomasdavis** [comment]
Great work! It looks like the concensus is that `keywords` should make it into the `projects` section.

Seems pretty good to me (y) 

Just to stall a tiny bit, are there any other synonyms for `keywords` just for the sake of comparing options. (I can't think of any)

**Comment by indrif** [comment]
What more is needed to settle this PR? I would really like to see the `projects` section added to the schema. If there is anything I can do to help let me know.

To me it seems that both `tech` and `team` properties could either be removed or tweaked to fill a more general need. Tech seems to be a very specific use case and team is something that can be described in the `summary`. There is no `team` in any other sections from what I know and I would really like to see the different sections as homogenous as possible.

**Comment by ciderpunx** [comment]
I would also like to see projects become part of the schema.

**Comment by stp-ip** [comment]
The latest changes look good. I will review the PR and reread the whole discussion on projects and will get back to you as soon as I can. (This is high priority for today, even if I have to skip some sleep ;) )
I really appreciate the work.

**Comment by ashes74** [comment]
I see the request to add `roles` to the project section of the schema. 

Do we want it as an array or an object?

**Comment by stp-ip** [comment]
With role the idea within the work section was to use a simple string. I think that also makes sense for projects.
So "role": "attendee" or "role": "Product Lead".

Any reason you would wanna use an array?

**Comment by stp-ip** [comment]
I now get what you wanted to do with the array for roles: https://github.com/jsonresume/resume-schema/issues/226#issuecomment-187022391

File: examples/valid/complete.json [hunk 1]
```diff
@@ -127,6 +127,23 @@
...
+         "Won award at AIHacks 2016",
+         "Built by all women team of newbie programmers",
+         "Using modern technologies such as GoogleMaps, Chrome Extension and Javascript"
+       ],
+       "keywords": [
+         "GoogleMaps", "Chrome Extension", "Javascript"
+       ],
+       "startDate": "2016-08-24",
+       "endDate": "2016-08-24", 
+       "website": "missdirection.io"
```
Comment (stp-ip) [inline examples/valid/complete.json]: Use ```url``` for now as in #256 and ```missdirection.example.com``` to prevent issues.

File: schema.json [hunk 2]
```diff
@@ -370,6 +370,57 @@
...
+      "additionalItems": false,
+      "items": {
+        "type": "object",
+        "additionalProperties": true,
+        "properties": {
+          "name": {
+            "type": "string",
+            "description": "e.g. The World Wide Web"
+          },
+          "summary": {
```
Comment (stp-ip) [inline schema.json]: Use ```description``` for now as in #256.

File: schema.json [hunk 3]
```diff
@@ -370,6 +370,57 @@
...
+          },
+          "endDate": {
+            "type": "string",
+            "description": "e.g. 2012-06-29",
+            "format": "date"
+          },
+          "website": {
+            "type": "string",
+            "description": "e.g. http://www.computer.org/csdl/mags/co/1996/10/rx069-abs.html"
+          }
```
Comment (stp-ip) [inline schema.json]: Add "format": "uri"

**Comment by stp-ip** [comment]
So again thanks for the changes.

I commented inline a bit and some general thoughts:

Arrays for roles seems the more generic solution so +1 for an array.
To make current changes easier I think updating the html is not feasible. I would say for new PRs we leave it to a complete rewrite after the schema stabilizes. Probably best to either remove it for now or we submit it knowing, that it's out of date.

**Comment by stp-ip** [comment]
Additionally add an ```entity``` field for projects. I think it is enough to have that as string only for now.
Examples of entities: jsonresume, greenpeace etc

**Comment by stp-ip** [comment]
I added all items left to do to the PR description so you can check them off.

**Comment by ashes74** [comment]
Completed all tasks related to feedback. Let me know, if anything else pops up.

**Comment by stp-ip** [comment]
Sorry for getting back to you so late. I was off the grid on holidays. Will take a look at your changes tomorrow. Thanks for the effort already.

**Comment by ashes74** [comment]
Thanks Michael

**Comment by stp-ip** [comment]
Awesome work. LGTM

**Comment by aloisdg** [comment]
\o/

**Comment by elupus** [comment]
Is there some theme:s that make use of this?

**Comment by stp-ip** [comment]
Not aware of one currently.

**Comment by rbardini** [comment]
[jsonresume-theme-even](https://github.com/rbardini/jsonresume-theme-even) supports projects. You can see an example [here](https://github.com/rbardini/resume.rbardini.com).

**Pull request merged**